### PR TITLE
[FLINK-17857][test] Make K8s e2e tests could run on Mac

### PR DESCRIPTION
**Pull request by wangyang0918**
Since the code in master has changes after #12451 is approved, I am afraid some K8s e2e tests still could not work on Mac OS. This PR will fix these tests and make them could run on Mac. Of cause, the PR will not affect the E2E tests on azure.

I have run the following tests on Mac for more than five times for each. They could all pass.

* `test_kubernetes_application.sh`
* `test_kubernetes_embedded_job.sh`
* `test_kubernetes_session.sh`
* `test_kubernetes_itcases.sh`
* `test_kubernetes_pyflink_application.sh`

### CODE REVIEW

**Comment by wangyang0918** [comment]
cc @rmetzger could you please have a look at your convenience?

File: flink-end-to-end-tests/test-scripts/common_kubernetes.sh [hunk 1]
```diff
@@ -25,7 +25,7 @@ MINIKUBE_START_RETRIES=3
 MINIKUBE_START_BACKOFF=5
 RESULT_HASH="e682ec6622b5e83f2eb614617d5ab2cf"
 MINIKUBE_VERSION="v1.8.2"
-MINIKUBE_PATH="/usr/local/bin/minikube-$MINIKUBE_VERSION"
+MINIKUBE_PATH=$([ "${OS_TYPE}" != "linux" ] && echo "minikube" || echo "/usr/local/bin/minikube-$MINIKUBE_VERSION")
```
Comment (tillrohrmann) [inline flink-end-to-end-tests/test-scripts/common_kubernetes.sh:28]: Why is this change necessary @wangyang0918?

File: flink-end-to-end-tests/test-scripts/common_kubernetes.sh [hunk 2]
```diff
@@ -178,21 +180,9 @@ function cleanup {
...
-    cat >> $FLINK_DIR/conf/log4j.properties <<END
-rootLogger.appenderRef.console.ref = ConsoleAppender
-
-# Log all infos to the console
-appender.console.name = ConsoleAppender
-appender.console.type = CONSOLE
-appender.console.layout.type = PatternLayout
-appender.console.layout.pattern = %d{yyyy-MM-dd HH:mm:ss,SSS} %-5p [%t] %-60c %x - %m%n
-END
-}
```
Comment (tillrohrmann) [inline flink-end-to-end-tests/test-scripts/common_kubernetes.sh:191]: Why don't we need this logging anymore?

**Review by tillrohrmann — COMMENTED** [review]
Thanks for creating this PR @wangyang0918. I had a couple of questions. Please take a look.

File: flink-end-to-end-tests/test-scripts/common_kubernetes.sh (hunk 1 above)
Comment (wangyang0918) [reply flink-end-to-end-tests/test-scripts/common_kubernetes.sh:28]: In linux environment, we are downloading the specific minikube version(e.g. v1.8.2) and set the correct `MINIKUBE_PATH`. However, on Mac, we assume that the minikube is pre-installed and we could directly use the `minikube` command to check status or mounting.

File: flink-end-to-end-tests/test-scripts/common_kubernetes.sh (hunk 2 above)
Comment (wangyang0918) [reply flink-end-to-end-tests/test-scripts/common_kubernetes.sh:191]: First, `setConsoleLogging` is not used in any native K8s related test scripts. Second, the native K8s is using `flink-console.properties` from release 1.12, which will output logs to console and files simultaneously.

File: flink-end-to-end-tests/test-scripts/common_kubernetes.sh (hunk 1 above)
Comment (wangyang0918) [reply flink-end-to-end-tests/test-scripts/common_kubernetes.sh:28]: Currently, when running native K8s related test on Mac, we will get following errors.
```
./test-scripts/common_kubernetes.sh: line 55: /usr/local/bin/minikube-v1.8.2: No such file or directory
****** Please start/stop minikube manually in non-linux environment. ******
```

**Comment by wangyang0918** [comment]
Rebase latest master and force pushed.

File: flink-end-to-end-tests/test-scripts/common_kubernetes.sh (hunk 1 above)
Comment (tillrohrmann) [reply flink-end-to-end-tests/test-scripts/common_kubernetes.sh:28]: Is `v1.8.2` strictly required? If yes, then we might wanna check on MacOS that `minikube` is actually the correct version. I for example have `1.15.1` installed.

File: flink-end-to-end-tests/test-scripts/common_kubernetes.sh (hunk 2 above)
Comment (tillrohrmann) [reply flink-end-to-end-tests/test-scripts/common_kubernetes.sh:191]: I noticed that we don't log any information when running `KubernetesLeaderElectionAndRetrievalITCase`. Could it be that we are using `log4j-test.properties` when running these tests because we use maven? Maybe we should set the log level to `INFO` in order to see the logs. Alternatively, we would have to use a log4j file which configures a file for the output which is then uploaded when the e2e fail (similarly to what we do when running the unit/integration tests which write into `mvn-1.log` and `mvn-2.log`. I think we use https://github.com/apache/flink/blob/master/tools/ci/log4j.properties.

Maybe we should exactly do this and use https://github.com/apache/flink/blob/master/tools/ci/log4j.properties for the logging configuration for the e2e tests which use maven to run them.

File: flink-end-to-end-tests/test-scripts/common_kubernetes.sh (hunk 1 above)
Comment (wangyang0918) [reply flink-end-to-end-tests/test-scripts/common_kubernetes.sh:28]: Hmm. `v1.8.2` is not strictly required. We are using this version just because it is the latest stable version when we introduce the tests.

File: flink-end-to-end-tests/test-scripts/common_kubernetes.sh (hunk 2 above)
Comment (wangyang0918) [reply flink-end-to-end-tests/test-scripts/common_kubernetes.sh:191]: I will have a look on the logging of unit test and try to correct the logging configuration for e2e tests that use `run_mvn`.

File: flink-end-to-end-tests/test-scripts/common_kubernetes.sh (hunk 1 above)
Comment (tillrohrmann) [reply flink-end-to-end-tests/test-scripts/common_kubernetes.sh:28]: Then we could maybe change it so that we use whatever version is available and if `minikube` is not available, then we try to download minikube on Linux. On MacOS we fail with a descriptive message. This would be a somewhat consistent behaviour.

Comment (wangyang0918) [reply flink-end-to-end-tests/test-scripts/common_kubernetes.sh:28]: I remember the azure pipeline updated the provided minikube version silently(a not stable version) and that leads to all the K8s related tests failed. This is why we have pinned the minikube version to `v1.8.2`.

Comment (tillrohrmann) [reply flink-end-to-end-tests/test-scripts/common_kubernetes.sh:28]: Would it make sense to check whether there is a `minikube` version and if whether it is a version which works? In this case we could use it. If not, then the system could try to download and install `minikube` `v1.8.2` if it runs on Linux.

**Comment by wangyang0918** [comment]
cc @tillrohrmann I have updated this PR.

This PR should rebase the master after #15849 is merged. Otherwise, the `test_kubernetes_itcases.sh` will fail.
